### PR TITLE
Extract breadcrumb capture to separate classes

### PR DESCRIPTION
**Pull request by fractalwrench**
## Goal

Refactors breadcrumb service to break the breadcrumb capture into separate classes for data capture. This will ultimately make it easier to have these classes implement `DataSource`

## Testing

Relied on existing unit test coverage.

### CODE REVIEW

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/capture/crumbs/BreadcrumbService.kt [hunk 1]
```diff
@@ -31,11 +31,9 @@ internal interface BreadcrumbService {
      * Gets the Taps breadcrumbs in the specified time window.
      * If the number of elements exceeds the limit, this will return the newest (latest) ones.
      *
-     * @param start the start time
-     * @param end   the end time
      * @return the list of Breadcrumbs
      */
-    fun getTapBreadcrumbsForSession(start: Long, end: Long): List<TapBreadcrumb?>
+    fun getTapBreadcrumbsForSession(): List<TapBreadcrumb>
```
Comment (bidetofevil) [inline embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/capture/crumbs/BreadcrumbService.kt:36]: lol finally getting rid of the interval 🫶

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/capture/crumbs/EmbraceBreadcrumbService.kt [hunk 2]
```diff
@@ -455,13 +313,13 @@ internal class EmbraceBreadcrumbService(
 
     override fun cleanCollections() {
         viewBreadcrumbs.clear()
```
Comment (bidetofevil) [inline embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/capture/crumbs/EmbraceBreadcrumbService.kt:315]: Will these be moved as well later or are they staying in this service for a reason?

**Review by bidetofevil — APPROVED** [review]
Looks good!

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/capture/crumbs/EmbraceBreadcrumbService.kt (hunk 2 above)
Comment (fractalwrench) [reply embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/capture/crumbs/EmbraceBreadcrumbService.kt:315]: These are a little bit harder to grok - breadcrumbs are removed from the queue after being added & IMO they should probably be modelled as spans instead. I'll extract these in a future PR.

**Pull request merged**